src/expr.rs

```diff
@@ -0,0 +1,15 @@
+use crate::token::token::Token;
+
+pub trait Expression {}
+
+pub struct Expr {
+    // I don't fully understand why this works with a `Box<>` and fails without.
+    pub left: Option<Box<Expr>>,
+    pub operator: Option<&'static Token>,
```

Comment:
It's odd that this has a `&'static` lifetime. That effectively says that `Token` lives for the entire life of the program, and I'm not sure why that's needed. 

What happens if you remove it entirely?

```suggestion
    pub operator: Option<Token>,
```

Reply:
Yeah this is me just trying stuff from online to solve errors in the compiler, I forget why I needed a lifetime here and I think I started with `'a` but tried `'static` when that didn't work.

I think it's probably just easy to remove.